### PR TITLE
CVE-2022-82889

### PR DESCRIPTION
**Pull request by cyb3rjy0t**

### CODE REVIEW

File: rules/web/web_cve_2022_42889_text4shell_exploit.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+title: Text4Shell Exploit CVE-2022-42889
+id: 85d466b0-d74c-4514-84d3-2bdd3327588b
```
Comment (frack113) [inline rules/web/web_cve_2022_42889_text4shell_exploit.yml:2]:
```suggestion
id: ddfabcf8-fd9a-4fc9-ac95-8af965171171
```
duplicate id

File: rules/web/web_cve_2022_42889_text4shell_exploit.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+detection:
+    selection1:
+        cs-uri-query|contains|all:
+            - 'getRuntime%28%29'
+            - 'exec%28'
+    selection2:
+        cs-uri-query|contains|all: 
+            - 'getRuntime()'
+            - 'exec('  
+    condition: selection1 OR selection2
```
Comment (frack113) [inline rules/web/web_cve_2022_42889_text4shell_exploit.yml:24]:
```suggestion
    condition: selection1 or selection2
```
or is lowercase

File: rules/web/web_cve_2022_42889_text4shell_exploit.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+title: Text4Shell Exploit CVE-2022-42889
```
Comment (frack113) [inline rules/web/web_cve_2022_42889_text4shell_exploit.yml:1]: It is not realy the Exploit but a Payload that is detected.
For the exploit , we need the server log and find java run a curl.

**Comment by frack113** [comment]
As it is a java web payload it is better to update the rule [web_java_payload_in_access_logs](https://github.com/SigmaHQ/sigma/blob/master/rules/web/web_java_payload_in_access_logs.yml)

File: rules/web/web_cve_2022_42889_text4shell_exploit.yml (hunk 2 above)
Comment (nasbench) [reply rules/web/web_cve_2022_42889_text4shell_exploit.yml]: Resolved by modifying the rule mentioned in the comment.

**Comment by nasbench** [comment]
As stated by @frack113 above the rule you proposed fits the logic described in [web_java_payload_in_access_logs](https://github.com/SigmaHQ/sigma/blob/master/rules/web/web_java_payload_in_access_logs.yml) best. Since the exploit code you added is more of a generic exploit code rather than one specific to that CVE. It's best to add it to the other one.

I did that and deleted your rule. I also added you as an author to the other rule :)

Thanks for your contribution.

**Review by nasbench — APPROVED** [review]

**Review by frack113 — APPROVED** [review]

**Pull request merged**